upup/pkg/fi/fitasks/keypair.go

```diff
@@ -180,15 +185,22 @@ func (_ *Keypair) Render(c *fi.Context, a, e, changes *Keypair) error {
 			createCertificate = true
 		} else if changes.Type != "" {
 			createCertificate = true
+		} else if a.KeypairType == "" {
+			// The a.KeysetAPIVersion will == "" when the keyset is a legacy keyset and does not have a keypair.yaml
+			// file store in the state store
+			createCertificate = true
 		} else {
 			glog.Warningf("Ignoring changes in key: %v", fi.DebugAsJsonString(changes))
 		}
 	}
 
+	// TODO: I am wondering if we reuse the certificate information when a.KeysetAPIVersion == "" rather than
+	// TODO: issuing the new cert
```

Comment:
I am seeing two private pems in the keypair.yaml file.  Any idea how we can re-use the pem?

Reply:
I think that we are eventually calling https://github.com/kubernetes/kops/blob/master/pkg/pki/csr.go#L94

And that is creating the new cert.

Reply:
https://github.com/kubernetes/kops/blob/0b0932b493bace5e695760bf5e5e26c82b560a76/pkg/pki/csr.go#L94